cpp/src/Slice/Util.h

```diff
@@ -69,8 +69,11 @@ ciequals(const std::string& a, const std::string& b);
 std::string
 prependA(const std::string& s);
 
-TypePtr
-unwrapIfOptional(const TypePtr&);
+TypePtr unwrapIfOptional(const TypePtr& type);
+TypePtr unwrapIfAlias(const TypePtr& type);
```

Comment:
Type aliases are "transparent", meaning when we generate code for a sequence<T> or T member, where T is an alias, we only see the actual underlying type. As a result, `unwrapIfAlias` is not really useful. When generating code for aliases, surely we don't need to auto unwrap them.

Reply:
Unfortunately there are places where this is useful. Not many, there are some places in the language-specific compilers where we do type-lookup manually (by calling `unit->lookupXXX()`), which will get the actual type-alias and won't perform any automatic unwrapping (we keep the original type-aliases around in case we want to generate `using` statements for them).

I removed a few of these places in my PR, but there were others that were less trivial to fix, where these helpers were useful to perform the check and unwrap in a single line.